src/course-unit/add-component/AddComponent.jsx

```diff
@@ -2,30 +2,56 @@ import PropTypes from 'prop-types';
 import { useSelector } from 'react-redux';
 import { useNavigate } from 'react-router-dom';
 import { useIntl } from '@edx/frontend-platform/i18n';
-import { Button } from '@edx/paragon';
+import { useToggle } from '@edx/paragon';
 
 import { getCourseSectionVertical } from '../data/selectors';
 import { COMPONENT_ICON_TYPES } from '../constants';
-import ComponentIcon from './ComponentIcon';
+import ComponentModalView from './add-component-modals/ComponentModalView';
+import AddComponentButton from './add-component-btn';
 import messages from './messages';
 
-const AddComponent = ({ blockId, handleCreateNewCourseXblock }) => {
+const AddComponent = ({ blockId, handleCreateNewCourseXBlock }) => {
   const navigate = useNavigate();
   const intl = useIntl();
+  const [isOpenAdvanced, openAdvanced, closeAdvanced] = useToggle(false);
+  const [isOpenHtml, openHtml, closeHtml] = useToggle(false);
+  const [isOpenOpenAssessment, openOpenAssessment, closeOpenAssessment] = useToggle(false);
   const { componentTemplates } = useSelector(getCourseSectionVertical);
 
-  const handleCreateNewXblock = (type) => () => {
+  const handleCreateNewXBlock = (type, moduleName) => {
     switch (type) {
     case COMPONENT_ICON_TYPES.discussion:
     case COMPONENT_ICON_TYPES.dragAndDrop:
-      handleCreateNewCourseXblock({ type, parentLocator: blockId });
+      handleCreateNewCourseXBlock({ type, parentLocator: blockId });
       break;
     case COMPONENT_ICON_TYPES.problem:
     case COMPONENT_ICON_TYPES.video:
-      handleCreateNewCourseXblock({ type, parentLocator: blockId }, ({ courseKey, locator }) => {
+      handleCreateNewCourseXBlock({ type, parentLocator: blockId }, ({ courseKey, locator }) => {
         navigate(`/course/${courseKey}/editor/${type}/${locator}`);
       });
       break;
+    case COMPONENT_ICON_TYPES.library:
+      handleCreateNewCourseXBlock({ type, category: 'library_content', parentLocator: blockId });
+      break;
+    case COMPONENT_ICON_TYPES.advanced:
+      handleCreateNewCourseXBlock({
+        type: moduleName, category: moduleName, parentLocator: blockId,
```

Comment:
[option] You can simplify params to `type: moduleName, parentLocator: blockId,` because we use `type` value for `category` key as a backup value `category: category || type,` https://github.com/raccoongang/frontend-app-course-authoring/blob/ts-develop/src/course-unit/data/api.js#L109

Reply:
Corrected, thanks